src/protocols/bdx/BdxTransferSession.h

```diff
@@ -146,39 +147,33 @@ class DLL_EXPORT TransferSession
 
     /**
      * @brief
-     *   Indicates the presence of pending output and includes any data for the caller to take action on.
-     *
-     *   This method should be called frequently in order to be notified about any messages received. It should also be called after
-     *   most other methods in order to notify the user of any message that needs to be sent, or errors that occurred internally.
-     *
-     *   It is possible that consecutive calls to this method may emit different outputs depending on the state of the
-     *   TransferSession object.
-     *
-     *   Note that if the type outputted is kMsgToSend, the caller is expected to send the message immediately, and the session
-     *   timeout timer will begin at curTime.
+     *   Callback that the session calls to notify the caller about any messages received or messages that need to be send
+     *   or errors that occurred internally.
      *
      *   See OutputEventType for all possible output event types.
      *
-     * @param event     Reference to an OutputEvent struct that will be filled out with any pending output data
-     * @param curTime   Current time
+     * @param event     Reference to an OutputEvent struct that will be filled out with
+     *                  the generated output event
      */
-    void PollOutput(OutputEvent & event, System::Clock::Timestamp curTime);
+    using OutputEventCallback = void (*)(void * context, OutputEvent & event);
 
     /**
      * @brief
-     *   Initializes the TransferSession object and prepares a TransferInit message (emitted via PollOutput()).
+     *   Initializes the TransferSession object and prepares a TransferInit message.
      *
      *   A TransferSession object must be initialized with either StartTransfer() or WaitForTransfer().
      *
      * @param role      Inidcates whether this object will be sending or receiving data
      * @param initData  Data for initializing this object and for populating a TransferInit message
      *                  The role parameter will determine whether to populate a ReceiveInit or SendInit
-     * @param timeout   The amount of time to wait for a response before considering the transfer failed
+     * @param callback  OutputEventCallback that is to be registered with the transfer session and gets called when Output events
+     *                  are generated.
+     * @param context   Reference to the caller of this method
```

Comment:
Does it have to be the caller?  Or is this just opaque context that will be passed to `callback`?  Seems like the latter.  And this should:

1. Document that this is what `context` is used for: calling `callback`.
2. Document the point until which that can happen (i.e. when it's safe to free `context`).

Reply:
opaque context that will be passed to callback. will document